src/reducers/members.js

```diff
@@ -162,6 +162,126 @@ function onGetStatsDone(state, { error, payload }) {
   };
 }
 
+/**
+ * Inits the loading of member stats history.
+ * @param {Object} state
+ * @param {Object} action
+ * @return {Object} New state.
+ */
+function onGetStatsHistoryInit(state, action) {
+  const { handle, uuid } = action.payload;
+  let res = state[handle];
+  res = res ? _.clone(res) : {};
+  res.statsHistory = { loadingUuid: uuid };
+  return {
+    ...state,
+    [handle]: res,
+  };
+}
+
+/**
+ * Finalizes the loading of member stats history.
+ * @param {Object} state
+ * @param {Object} action
+ * @return {Object} New state.
+ */
+function onGetStatsHistoryDone(state, { error, payload }) {
+  if (error) {
+    logger.error('Failed to get member statsHistory', payload);
+    fireErrorMessage('Failed to get member statsHistory', '');
+    return state;
+  }
+
+  const { data, handle, uuid } = payload;
+  if (uuid !== _.get(state[handle], 'statsHistory.loadingUuid')) return state;
+  return {
+    ...state,
+    [handle]: {
+      ...state[handle],
+      statsHistory: { data, timestamp: Date.now() },
```

Comment:
I know, I did it myself in many places, but it turns out that setting timestamps with `Date.now()` inside reducer is anti-pattern: reducers should be pure functions, timestamp should created inside the action :)